src/cc65/scanner.c

```diff
@@ -575,16 +582,28 @@ static void NumericConst (void)
     /* Since we now know the correct base, convert the input into a number */
     SB_SetIndex (&Src, Index);
     IVal = 0;
+    Overflow = 0;
     while ((C = SB_Peek (&Src)) != '\0' && (Base <= 10 ? IsDigit (C) : IsXDigit (C))) {
         DigitVal = HexVal (C);
         if (DigitVal >= Base) {
             Error ("Invalid digit \"%c\" beyond radix %u constant", C, Base);
             SB_Clear (&Src);
             break;
         }
-        IVal = (IVal * Base) + DigitVal;
+        if (((scan_t)(IVal * Base) / Base) != IVal) {
+            Overflow = 1;
+        }
+        IVal = IVal * Base;
+        if (((scan_t)(IVal + DigitVal)) < IVal) {
+            Overflow = 1;
+        }
+        IVal += DigitVal;
         SB_Skip (&Src);
     }
+    if (Overflow) {
+        Error ("Numerical constant \"%s\" too large for internal %d-bit representation",
+               SB_GetConstBuf (&Src), (int)(sizeof(IVal)*8));
```

Comment:
I don't think that's the real problem. We could still do useful arithmetic with 64-bit integer constants if they were supported, regardless of not being able to store them. A similar story with floating points, though we don't support arithmetic with them yet either.

Reply:
As long as ExprDesc.IVal is a `long`, there's no way to do 64-bit arithmetic.

Reply:
long is 64-bit in the linux build

Reply:
Yeah, but not in windows. This would mean that cc65 would behave differently on different platforms. Something I would try to avoid.

Reply:
Yes, this is why I would ultimately suggest using an explicitly sized type for Expr.IVal and then follow the chain everywhere it goes, but this would be a rather large review and change. For now, to make the added error check consistent (which is necessary to pass the unit tests on both platforms) I can at least do this much with the numerical scanner.